terraform/modules/origin/main.tf

```diff
@@ -1,97 +1,161 @@
 locals {
-  mode = var.live ? "www" : "draft"
+  mode          = var.live ? "www" : "draft"
+  origin_alb_id = "${local.mode}_origin_alb"
+  origin_s3_id  = "origin_s3"
 }
 
-resource "aws_lb" "origin" {
-  name               = "${local.mode}-origin-ecs-${var.workspace}"
-  internal           = false
-  load_balancer_type = "application"
-  security_groups    = [aws_security_group.origin_alb.id]
-  subnets            = var.public_subnets
-}
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.33"
+    }
 
+    random = {
+      source  = "hashicorp/random"
+      version = "3.0.0"
+    }
+  }
+}
 
-resource "aws_lb_target_group" "origin-frontend" {
-  name        = "${local.mode}-origin-frontend-${var.workspace}"
-  port        = 80
-  protocol    = "HTTP"
-  vpc_id      = var.vpc_id
-  target_type = "ip"
+provider "aws" {
+  region = var.aws_region
 
-  health_check {
-    path = "/"
+  assume_role {
+    role_arn = var.assume_role_arn
   }
 }
 
-resource "aws_lb_listener_rule" "origin-frontend" {
-  listener_arn = aws_lb_listener.origin.arn
-  priority     = 100
+provider "aws" {
+  region = "us-east-1"
+  alias  = "use1"
 
-  action {
-    type             = "forward"
-    target_group_arn = aws_lb_target_group.origin-frontend.arn
+  assume_role {
+    role_arn = var.assume_role_arn
   }
+}
+
+resource "aws_cloudfront_origin_access_identity" "cloudfront_s3_access" {
+  comment = "${local.mode}-origin ${var.workspace} cloudfront accessing the Rails assets s3 bucket"
+}
+
+resource "random_password" "origin_alb_x_custom_header_secret" {
+  length  = 128
+  special = false
+}
+
+resource "aws_secretsmanager_secret" "origin_alb_x_custom_header_secret" {
+  name = "${local.mode}_origin_${var.workspace}_alb_x_custom_header_secret"
+}
+
+resource "aws_secretsmanager_secret_version" "origin_alb_x_custom_header_secret" {
+  secret_id     = aws_secretsmanager_secret.origin_alb_x_custom_header_secret.id
+  secret_string = random_password.origin_alb_x_custom_header_secret.result
+}
+
+resource "aws_cloudfront_distribution" "origin" {
+  origin {
+    domain_name = aws_lb.origin.dns_name
+    origin_id   = local.origin_alb_id
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "https-only"
+      origin_ssl_protocols   = ["TLSv1.2"]
+    }
 
-  condition {
-    path_pattern {
-      values = ["/assets/frontend/*"]
+    #TODO: frequent rototation for additional security,
```

Comment:
I suspect that this is a TODO which will never get prioritized. Could you raise a trello card for it, and maybe link to that here?

Something like "Automate the rotation of the X-Cloudfront-Token secret"?

Reply:
thanks, I can do that.